From 06cc0a9f23100d7fede283e04b600af1627d9eed Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Fri, 14 Sep 2018 21:32:36 +0530
Subject: [PATCH 0193/1051] net: octeontx2: Fix next subdescriptor offset in
 SQE

Fixes next subdescriptor offset or size of SQE when
number of buffers in a SG descriptor is either 2 or 3.

Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c b/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
index cdfc8aea760d..eacb5c6853e4 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_txrx.c
@@ -400,7 +400,7 @@ static bool otx2_sqe_add_sg(struct otx2_nic *pfvf, struct otx2_snd_queue *sq,
 			 * So if sg->segs is whether 2 or 3, offset += 16bytes.
 			 */
 			if ((num_segs - seg) >= (MAX_SEGS_PER_SG - 1))
-				*offset += sizeof(*sg) + (2 * sizeof(u64));
+				*offset += sizeof(*sg) + (3 * sizeof(u64));
 			else
 				*offset += sizeof(*sg) + sizeof(u64);
 		}
-- 
2.17.1

